sklearn/linear_model/_ridge.py

```diff
@@ -1956,7 +1960,7 @@ def _solve_svd_design_matrix(self, alpha, y, sqrt_sw, X_mean, singvals_sq, U, UT
             G_inverse_diag = G_inverse_diag[:, np.newaxis]
         return G_inverse_diag, c
 
-    def fit(self, X, y, sample_weight=None):
+    def fit(self, X, y, sample_weight=None, score_params=None):
```

Comment:
@adrinjalali since the class is private, do you think it would be better to have `def fit(self, X, y, **params)` and expect that we already packed the `sample_weight` in it before the call?